### PR TITLE
[FIX] Enable classification tests

### PR DESCRIPTION
**Pull request by JakaKokosar**
##### Issue

Classification tests did not run.

### CODE REVIEW

**Comment by JakaKokosar** [comment]
Tests fail with:

```
======================================================================
ERROR: test_larger_model (test_base.TestModelMapping)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/tests/test_base.py", line 48, in test_larger_model
    vale[vale != 0] - 1)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 931, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

Mismatched elements: 100 / 100 (100%)
Max absolute difference: 1
Max relative difference: 1.
 x: array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2,...
 y: array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,...

======================================================================
ERROR: test_model_different (test_base.TestModelMapping)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/tests/test_base.py", line 138, in test_model_different
    np.testing.assert_array_equal(prob1[valp0 == 1, 1], 1)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 931, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

Mismatched elements: 48 / 48 (100%)
Max absolute difference: 0.41354173
Max relative difference: 0.41354173
 x: array([0.999999, 0.999998, 0.999999, 0.999998, 0.999999, 0.999997,
       0.999998, 0.999998, 0.999998, 0.999998, 0.999999, 0.999997,
       0.999999, 0.999999, 1.      , 0.999999, 0.999999, 0.999999,...
 y: array(1)

======================================================================
ERROR: test_smaller_model (test_base.TestModelMapping)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/tests/test_base.py", line 89, in test_smaller_model
    np.testing.assert_array_equal(val0, val[50:] - 1)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 931, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

Mismatched elements: 100 / 100 (100%)
Max absolute difference: 1.
Max relative difference: 1.
 x: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 2., 1., 1., 1., 1., 1., 1., 2., 1., 1., 1., 1., 1., 2.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 2.,...
 y: array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 1.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.,...

======================================================================
ERROR: test_fit_storage (test_calibration.TestCalibratedLearner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/unittest/mock.py", line 1183, in patched
    return func(*args, **keywargs)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/tests/test_calibration.py", line 190, in test_fit_storage
    cal_model = cal_learner(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/base.py", line 136, in __call__
    model = self._fit_model(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/base.py", line 148, in _fit_model
    return self.fit_storage(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/calibration.py", line 164, in fit_storage
    model = res.models[0, 0]
TypeError: 'Mock' object is not subscriptable

======================================================================
ERROR: test_fit_storage (test_calibration.TestThresholdLearner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/unittest/mock.py", line 1183, in patched
    return func(*args, **keywargs)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/tests/test_calibration.py", line 89, in test_fit_storage
    thresh_model = thresh_learner(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/base.py", line 136, in __call__
    model = self._fit_model(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/base.py", line 148, in _fit_model
    return self.fit_storage(data)
  File "/home/runner/work/orange3/orange3/.tox/coverage/lib/python3.6/site-packages/Orange/classification/calibration.py", line 78, in fit_storage
    model = res.models[0, 0]
TypeError: 'Mock' object is not subscriptable
```

File: Orange/classification/tests/test_base.py [hunk 1]
```diff
@@ -12,17 +10,21 @@ class TestModelMapping(unittest.TestCase):
...
+        ix = iris.X
+        y = np.hstack((np.zeros(50), np.ones(50)))
+        attrs = cls.iris.domain.attributes
+        classes = cls.iris.domain.class_var.values
+        for i, x in enumerate([ix[50:],
+                               np.vstack((ix[:50], ix[100:])),
+                               ix[:100]]):
+            class_var = DiscreteVariable(
+                "iris",
+                values=tuple(n for j, n in enumerate(classes) if n != i))
```
Comment (markotoplak) [inline Orange/classification/tests/test_base.py]: Needs to be j != i

Comment (markotoplak) [reply Orange/classification/tests/test_base.py]: If this is fixed the other functions do not need changes. I added a fixup commit which returns them into their original state. @janezd, could you confirm if that makes sense?

**Comment by markotoplak** [comment]
@janezd, I simplified test changes: only the function generating test tables needed changes; I left the others as they were in master.

**Comment by janezd** [comment]
@markotoplak, thanks, your fix is the correct one.

I identified the problem in the setup, incorrectly fixed it, and then looked at failing tests for long enough to convince myself that this is how backmapping works like. :(

**Pull request merged**